main.c

```diff
@@ -0,0 +1,107 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <math.h>
+
+#include <openssl/sha.h>
+#include <string.h>
+
+void printa(unsigned char* array, int len) {
+	for (int i = 0; i < len; i++) {
+		printf("%3i, ", array[i]);
+	}
+	printf("\n");
+}
+
+void printx(unsigned char* array, int len) {
+	for (int i = 0; i < len; i++) {
+		printf("%X", array[i]);
+	}
+	printf("\n");
+}
+
+// Unsigned Char CoPY
+void uccpy(unsigned char* a, unsigned char* b, int n, int i) {
+	if (i < n) {
+		a[i] = b[i];
+		uccpy(a, b, n, ++i);
+	}
+}
+
+typedef struct {
+	unsigned char* result;
+	int length;
+	int remainder;
+} Result;
+
+// NOTE: Denominator has to be less than the base
+Result division(unsigned char* numerand, int length_of_numerand, int denominator, int base) {
```

Comment:
This is a function to get the division between a number in base n and a number in base 10.

SHA-512 returns an array of 64 bytes. We could consider these bytes as a number in base 256.

So, we use this function to divide the hash with any number that we would want.